src/sicmutils/generic.cljc

```diff
@@ -119,17 +114,15 @@
 (defmulti simplify v/argument-kind)
 
 (defn ^:private bin+ [a b]
-  (cond (and (number? a) (number? b)) (#?(:clj +' :cljs core-plus) a b)
```

Comment:
these first lines are all handled by the multimethods, assuming you import `sicmutils.numbers`.